lib/awful/hotkeys_popup/widget.lua

```diff
@@ -481,32 +484,28 @@ function widget.new(args)
 
         local function insert_keys(_keys, _add_new_column)
             local max_label_width = 0
-            local max_label_content = ""
             local joined_labels = ""
             for i, key in ipairs(_keys) do
-                local length = string.len(key.key or '') + string.len(key.description or '')
                 local modifiers = key.mod
                 if not modifiers or modifiers == "none" then
                     modifiers = ""
                 else
-                    length = length + string.len(modifiers) + 1 -- +1 for "+" character
                     modifiers = markup.fg(self.modifiers_fg, modifiers.."+")
                 end
                 local rendered_hotkey = markup.font(self.font,
                     modifiers .. (key.key or "") .. " "
                 ) .. markup.font(self.description_font,
                     key.description or ""
                 )
-                if length > max_label_width then
-                    max_label_width = length
-                    max_label_content = rendered_hotkey
+                local label_width, _ = wibox.widget.textbox(rendered_hotkey):get_preferred_size(s)
```

Comment:
Can you switch this to Pango code rather than make a disposable widget? There is plenty of examples in the `tests/example` directory (git-grep -i pango in there).